-1

Tested at: ddaef4ba14cbb2ff650b595838f102794e088cd0

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-24503/30339/summary.html

I found follow errors while testing this PR

**Failed tests:  Build  HeaderConsistency**

* **Build**:

I found an error when building: 
 
 <pre>>> Compiling edm plugin /build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_3_X_2018-09-10-2300/src/GeneratorInterface/RivetInterface/plugins/RivetHarvesting.cc 
>> Compiling edm plugin /build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_3_X_2018-09-10-2300/src/GeneratorInterface/RivetInterface/plugins/MergedGenParticleProducer.cc 
>> Compiling edm plugin /build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_3_X_2018-09-10-2300/src/GeneratorInterface/RivetInterface/plugins/RivetAnalyzer.cc 
>> Compiling edm plugin /build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_3_X_2018-09-10-2300/src/GeneratorInterface/RivetInterface/plugins/ParticleLevelProducer.cc 
>> Compiling edm plugin /build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_3_X_2018-09-10-2300/src/GeneratorInterface/RivetInterface/plugins/GenParticles2HepMCConverter.cc 
/build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_3_X_2018-09-10-2300/src/GeneratorInterface/RivetInterface/plugins/RivetHarvesting.cc:14:10: fatal error: tinyxml.h: No such file or directory
 #include "tinyxml.h"
          ^~~~~~~~~~~
compilation terminated.
>> Compiling edm plugin /build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_3_X_2018-09-10-2300/src/GeneratorInterface/RivetInterface/plugins/DQMRivetClient.cc 
/build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_3_X_2018-09-10-2300/src/GeneratorInterface/RivetInterface/plugins/RivetHarvesting.cc:14:10: fatal error: tinyxml.h: No such file or directory
</pre>

--- a/FWCore/MessageLogger/src/JobReport.cc
+++ b/FWCore/MessageLogger/src/JobReport.cc
@@ -243,7 +243,7 @@ namespace edm {
         *ost_ << "\n  <PFN>" << doc.NewText(inpFile.physicalFileName.c_str())->Value() << "</PFN>";
         *ost_ << "\n  <FastCopying>" << findOrDefault(f.fastCopyingInputs, inpFile.physicalFileName) << "</FastCopying>";
         *ost_ << "\n</Input>";
-	doc.DeleteChildren();
+        doc.DeleteChildren();
       }
       for(auto token : f.contributingInputsSecSource) {
         JobReport::InputFile inpFile = inputFilesSecSource_.at(token);
@@ -252,7 +252,7 @@ namespace edm {
         *ost_ << "\n  <PFN>" << doc.NewText(inpFile.physicalFileName.c_str())->Value() << "</PFN>";
         *ost_ << "\n  <FastCopying>" << findOrDefault(f.fastCopyingInputs, inpFile.physicalFileName) << "</FastCopying>";
         *ost_ << "\n</Input>";
-	doc.DeleteChildren();
+        doc.DeleteChildren();
       }
       *ost_ << "\n</Inputs>";
       *ost_ << "\n</File>\n";
@@ -762,7 +762,7 @@ namespace edm {
         msg << "\n  <PFN>" << doc.NewText(inpFile.physicalFileName.c_str())->Value() << "</PFN>";
         msg << "\n  <FastCopying>" << findOrDefault(f.fastCopyingInputs, inpFile.physicalFileName) << "</FastCopying>";
         msg << "\n</Input>";
-	doc.DeleteChildren();
+        doc.DeleteChildren();
       }
       msg << "\n</Inputs>";
       msg << "\n</File>";

--- a/FWCore/MessageLogger/src/JobReport.cc
+++ b/FWCore/MessageLogger/src/JobReport.cc
@@ -52,6 +52,7 @@ namespace edm {
     os << "\n<Branches>";
     for(auto const& branch : f.branchNames) {
       os << "\n  <Branch>" << doc.NewText(branch.c_str())->Value() << "</Branch>";
+      doc.DeleteChildren();
     }
     os << "\n</Branches>";
     return os;
@@ -242,6 +243,7 @@ namespace edm {
         *ost_ << "\n  <PFN>" << doc.NewText(inpFile.physicalFileName.c_str())->Value() << "</PFN>";
         *ost_ << "\n  <FastCopying>" << findOrDefault(f.fastCopyingInputs, inpFile.physicalFileName) << "</FastCopying>";
         *ost_ << "\n</Input>";
+	doc.DeleteChildren();
       }
       for(auto token : f.contributingInputsSecSource) {
         JobReport::InputFile inpFile = inputFilesSecSource_.at(token);
@@ -250,6 +252,7 @@ namespace edm {
         *ost_ << "\n  <PFN>" << doc.NewText(inpFile.physicalFileName.c_str())->Value() << "</PFN>";
         *ost_ << "\n  <FastCopying>" << findOrDefault(f.fastCopyingInputs, inpFile.physicalFileName) << "</FastCopying>";
         *ost_ << "\n</Input>";
+	doc.DeleteChildren();
       }
       *ost_ << "\n</Inputs>";
       *ost_ << "\n</File>\n";
@@ -759,6 +762,7 @@ namespace edm {
         msg << "\n  <PFN>" << doc.NewText(inpFile.physicalFileName.c_str())->Value() << "</PFN>";
         msg << "\n  <FastCopying>" << findOrDefault(f.fastCopyingInputs, inpFile.physicalFileName) << "</FastCopying>";
         msg << "\n</Input>";
+	doc.DeleteChildren();
       }
       msg << "\n</Inputs>";
       msg << "\n</File>";

--- a/GeneratorInterface/RivetInterface/BuildFile.xml
+++ b/GeneratorInterface/RivetInterface/BuildFile.xml
@@ -5,7 +5,6 @@
 <use name="boost"/>
 <use name="gsl"/>
 <use name="root"/>
-<use name="tinyxml"/>
 <use name="GeneratorInterface/Core"/>
 <use name="GeneratorInterface/Pythia6Interface"/>
 <flags   RIVET_PLUGIN="1"/>

--- a/GeneratorInterface/RivetInterface/plugins/BuildFile.xml
+++ b/GeneratorInterface/RivetInterface/plugins/BuildFile.xml
@@ -6,6 +6,7 @@
 <use   name="rivet"/>
 <use   name="yoda"/>
 <use   name="gsl"/>
+<use   name="tinyxml2"/>
 <use   name="DQMServices/Core"/>
 <use   name="SimGeneral/HepPDTRecord"/>
 <use   name="DataFormats/HepMCCandidate"/>

--- a/GeneratorInterface/RivetInterface/plugins/RivetHarvesting.cc
+++ b/GeneratorInterface/RivetInterface/plugins/RivetHarvesting.cc
@@ -11,7 +11,7 @@
 #include "Rivet/AnalysisHandler.hh"
 #include "Rivet/Analysis.hh"
 #include "Rivet/Tools/RivetYODA.hh"
-#include "tinyxml.h"
+#include "tinyxml2.h"
 
 #include <string>
 #include <vector>
@@ -22,6 +22,7 @@
 using namespace Rivet;
 using namespace edm;
 using namespace std;
+using namespace tinyxml2;
 
 RivetHarvesting::RivetHarvesting(const edm::ParameterSet& pset) : 
 _analysisHandler(),
@@ -186,11 +187,12 @@ void RivetHarvesting::endJob(){
 vector<YODA::Point2D> RivetHarvesting::getPoint2DValsErrs(std::string filename, std::string path, std::string name) {
   
     // Open YODA XML file
-    TiXmlDocument doc(filename);
-    doc.LoadFile();
+    XMLDocument doc;
+    doc.LoadFile(filename.c_str());
+    //doc.LoadFile();
     if (doc.Error()) {
-      string err = "Error in " + string(doc.Value());
-      err += ": " + string(doc.ErrorDesc());
+      string err = "Error in " + filename;
+      err += ": " + string(doc.ErrorStr());
       cerr << err << endl;
       throw cms::Exception("RivetHarvesting") << "Cannot open " << filename;
     }
@@ -200,10 +202,9 @@ vector<YODA::Point2D> RivetHarvesting::getPoint2DValsErrs(std::string filename, 
     
     try {
       // Walk down tree to get to the <paper> element
-      const TiXmlNode* yodaN = doc.FirstChild("yoda");
+      const XMLElement* yodaN = doc.FirstChildElement("yoda");
       if (!yodaN) throw cms::Exception("RivetHarvesting") << "Couldn't get <yoda> root element";
-      for (const TiXmlNode* dpsN = yodaN->FirstChild("dataPointSet"); dpsN; dpsN = dpsN->NextSibling()) {
-        const TiXmlElement* dpsE = dpsN->ToElement();
+      for (const XMLElement* dpsE = yodaN->FirstChildElement("dataPointSet"); dpsE; dpsE = dpsE->NextSiblingElement()) {
         const string plotname = dpsE->Attribute("name");
         const string plotpath = dpsE->Attribute("path");
         if (plotpath != path && plotname != name)
@@ -216,12 +217,10 @@ vector<YODA::Point2D> RivetHarvesting::getPoint2DValsErrs(std::string filename, 
 
         /// @todo Check that "path" matches filename
         vector<Point2D> points;
-        for (const TiXmlNode* dpN = dpsN->FirstChild("dataPoint"); dpN; dpN = dpN->NextSibling()) {
-          const TiXmlNode* xMeasN = dpN->FirstChild("measurement");
-          const TiXmlNode* yMeasN = xMeasN->NextSibling();
-          if (xMeasN && yMeasN)  {
-            const TiXmlElement* xMeasE = xMeasN->ToElement();
-            const TiXmlElement* yMeasE = yMeasN->ToElement();
+        for (const XMLElement* dpE = dpsE->FirstChildElement("dataPoint"); dpE; dpE = dpE->NextSiblingElement()) {
+          const XMLElement* xMeasE = dpE->FirstChildElement("measurement");
+          const XMLElement* yMeasE = xMeasE->NextSiblingElement();
+          if (xMeasE && yMeasE)  {
             const string xcentreStr   = xMeasE->Attribute("value");
             const string xerrplusStr  = xMeasE->Attribute("errorPlus");
             const string xerrminusStr = xMeasE->Attribute("errorMinus");

